packages/app-store/btcpayserver/components/EventTypeAppSettingsInterface.tsx

```diff
@@ -0,0 +1,135 @@
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { Alert } from "@calcom/ui/components/alert";
+import { Select } from "@calcom/ui/components/form";
+import { TextField } from "@calcom/ui/components/form";
+
+import {
+  currencyOptions,
+  convertToSmallestCurrencyUnit,
+  convertFromSmallestToPresentableCurrencyUnit,
+} from "../lib/currencyOptions";
+import { BTCPayPaymentOptions as paymentOptions } from "../zod";
+
+type Option = { value: string; label: string };
+
+const EventTypeAppSettingsInterface: EventTypeAppSettingsComponent = ({
+  eventType,
+  getAppData,
+  setAppData,
+}) => {
+  const { t } = useLocale();
+  const price = getAppData("price");
+  const currency = getAppData("currency") || (currencyOptions.length > 0 ? currencyOptions[0].value : "");
+  const [selectedCurrency, setSelectedCurrency] = useState(
+    currencyOptions.find((c) => c.value === currency) ||
+      (currencyOptions.length > 0
+        ? {
+            label: currencyOptions[0].label,
+            value: currencyOptions[0].value,
+          }
+        : null)
+  );
+
+  const paymentOption = getAppData("paymentOption");
+  const paymentOptionSelectValue = paymentOptions?.find((option) => paymentOption === option.value) || {
+    label: paymentOptions.length > 0 ? paymentOptions[0].label : "",
+    value: paymentOptions.length > 0 ? paymentOptions[0].value : "",
+  };
+  const seatsEnabled = !!eventType.seatsPerTimeSlot;
+  const [requirePayment, setRequirePayment] = useState(getAppData("enabled"));
+  const recurringEventDefined = eventType.recurringEvent?.count !== undefined;
+
+  // make sure a currency is selected
+  useEffect(() => {
+    if (requirePayment && !getAppData("currency")) {
+      setAppData("currency", currencyOptions[0].value);
+    }
+  }, [requirePayment, getAppData, setAppData]);
+
+  const disableDecimalPlace = (value: number) => {
+    return Math.floor(value);
+  };
+
+  return (
+    <>
+      {recurringEventDefined ? (
+        <Alert className="mt-2" severity="warning" title={t("warning_recurring_event_payment")} />
+      ) : (
+        requirePayment && (
+          <>
+            <div className="mt-4 block items-center justify-start sm:flex sm:space-x-2">
+              <TextField
+                label={t("price")}
+                className="block w-full rounded-sm border-gray-300 pl-2 pr-12 text-sm"
+                addOnClassname="h-[38px]"
+                step="1"
+                min="1"
+                type="number"
+                required
+                placeholder="Price"
+                onChange={(e) => {
+                  setAppData("price", convertToSmallestCurrencyUnit(Number(e.target.value), currency));
+                }}
+                value={
+                  price && price > 0
+                    ? disableDecimalPlace(convertFromSmallestToPresentableCurrencyUnit(price, currency))
+                    : undefined
+                }

```

Comment:
<img width="846" height="338" alt="image" src="https://github.com/user-attachments/assets/22789924-b0a0-46ed-a79b-a0b3a218d345" />
The UI looks wrong here, no rounded corners and misplaced arrows